### ISSUE TITLE
Fix handling of blaclisted IPs read from storage

### ISSUE THREAD

**Issue by mina86** [near/nearcore#6521]
I’ve recently submitted [a change](https://github.com/near/nearcore/pull/6503) which causes peers read from the database to become banned if their IP address matches entry in a blacklist. There are still two outstanding issues:

- [ ] The banned peers will be unbanned after some time. We need to figure out how this should interact with blaclisted addresses (do blacklisted peers never get unbanned? Should we just remove them from the PeerStore instead of banning them?) and implement that.
- [ ] IPv4 addresses can be mapped to IPv6 (as ::ffff::a.b.c.d) which means that on some systems it might be possible to trick the node to accepting connection from a blacklisted IP if it’s somehow presented as IPv6 address. We should change the blacklist implementation so that we always either canonicalize addresses or convert them to IPv6.

**Comment by mooori** [comment]
I'm interested in working on the second item. Is [`std::net::Ipv4Addr::to_ipv6_mapped()`](https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html#method.to_ipv6_mapped) a valid approach for the conversion?

**Comment by mina86** [comment]
Yep.

**Comment by mooori** [comment]
Is there a preference regarding the first item (never unban or remove)?

Removing the peer from `PeerStore` might make things easier, since then there is no need for extra logic to handle peers which are banned *and* blacklisted. It should be safe since new connections from blacklisted peers are rejected:
https://github.com/near/nearcore/blob/55d0e03b52d1eece08ce4ba91cc726c9134087e9/chain/network/src/peer_manager/peer_manager_actor.rs#L2081-L2085

**Comment by mina86** [comment]
Yeah, I think deleting the nodes from the database should be fine.
